### ISSUE TITLE
Design Intent - Accessibility Feedback - Iterate Innovate Run, Performance Dashboard

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#69445]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- In the meeting I mentioned that I would have a list of minor improvements to recommend for the page, which I'm listing below. But I also recommend going through the exercise of completing the [foundational accessibility tests we require prior to Staging Review](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review). That's a little backwards from what's typical, but doing a pre-review for accessibility issues may help you get a good handle on this page (and it may be nice to do it once now before you do it again when you reach your Staging Review). Definitely no requirement to do this or to submit the testing artifact at this point, just something that might be helpful.
- I am very, very glad to see your team tackle this page (and hopefully more products on VA.gov after this one)! Please don't hesitate to reach out with questions, either directly to me or via the #accessibility-help channel in DSVA Slack.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Consider:** Data visualization is not easy to make accessible, especially for screen reader users. Right now, the bar chart is coded  with a very wordy `aria-label` so that screen readers announce the chart as something like:
   
   ```
   This chart displays the total number of Logged in Users for the various VA services over the last 12 months: 11/2022 - 1519284, 12/2022 - 1606892, 1/2023 - 1828906, 2/2023 - 1631227, 3/2023 - 1842010, 4/2023 - 1688884, 5/2023 - 1728195, 6/2023 - 1827034, 7/2023 - 2017975, 8/2023 - 2436171, 9/2023 - 2013107, 10/2023 - 1941599, 
   ```
   
   This communicates one aspect of the data visualization, but it doesn't really capture the "at a glance" impression of the trend lines --- something like "the number of logged in users has been gradually increasing for the last year, with a large spike in August 2023." Even with the raw data made accessible, it would take a pretty heavy cognitive load for a blind user to hear that `aria-label` and get that "at a glance" impression. It's very difficult to automate that kind of text description, but it's an important function of the chart that's lost for screen reader users.
   
   My recommendation here (which may be out of scope for your work) would be to consider whether it's worthwhile to keep the bar charts at all. If you threw out the bar chart, is there a different way you could communicate both the raw data and the highlights that could be automated and expressed just with text? Or maybe the better question is: What is this page actually trying to communicate, and is there a better way to do that?
   
   The answer might be no! But I would recommend going through the design exercise, just in case you do find a better solution.
- **Must:** As noted above, the charts have an `aria-label` intended to provide something for screen readers to announce --- but the `aria-label` isn't actually announced by screen readers. I believe it's because the `canvas` doesn't have a `role="img"` on it, although that's worth doing follow-up testing to verify. Adding `role="img"` may interfere with the tooltip functionality of the chart, so if not `role="img"` then consider putting an `aria-hidden="true"` on the `canvas` (hiding the element from screen readers entirely) and then using `sr-only` text next to the chart to provide the screen reader-friendly version.
- **Must:** Tooltips that only appear on `:hover` aren't available to keyboard-only users (see [WCAG SC 2.1.1](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)), which means keyboard-only users can't see the month-to-month data that mouse users can access. Tooltips in general are a problematic pattern for accessibility. I would recommend either removing the tooltip functionality and providing the raw data in a table or some other element, or else following [Scott O'Hara's pattern for an accessible tooltip](https://github.com/scottaohara/a11y_tooltips) that can be opened and closed via keyboard interaction.
- **Must:** The interaction pattern around the buttons to change which graph is displayed (Logged In Users, User Satisfaction, Successful Transactions) poses a couple of accessibility problems. I don't have specific recommendations on how to fix them, unfortunately, and I think it will take some design work to explore a few alternatives.
   - For screen reader users, pressing a button leads to a "selected" announcement but doesn't result in any announcements to indicate that any action has taken place. Sighted users will get the visual cue of the chart changing, but without those visual cues you're left to wonder whether anything has happened. This could be resolved via focus management (clicking the button moves keyboard focus to the chart, where screen reader text is announced), or by some screen reader only announcement that the page has updated (eg. using `aria-live`), or by rethinking the interaction pattern entirely.
   - For voice command users, the visual appearance of the buttons doesn't communicate (1) that they're interactive, or (2) that they're buttons specifically. Voice command users typically won't have some of the interaction cues that mouse users have (color change on hover, mouse changes to pointer), and there are some differences in what commands they may use for something they think is a button versus a link or something else. With few exceptions, we encourage people building on VA.gov to practice material honesty to help with this --- if it's a button, it looks like a button and acts like a button.
- **Should:** As mentioned in the meeting, look for any opportunity you can to replace elements with design system components. I'm seeing [additional info](https://design.va.gov/components/additional-info), [tags](https://design.va.gov/components/tag), [cards](https://design.va.gov/components/card), and [tables](https://design.va.gov/components/table) as components to look at. Note that using the table component will also require tables have a title, which resolves another accessibility issue.
